### PR TITLE
Add support for compressed keys

### LINKED ISSUE #14812: Show compressed chat keys
### Description

Instead of the uncompressed key format, show Chat keys in the compressed format.

Screen Shot 2023-01-17 at 3 24 48 PM

Screen Shot 2023-01-17 at 3 24 59 PM

### LINKED ISSUE #14843: View profile option does nothing
# Bug Report

## Problem 
Regression from 20/01/2023 (there is no issue on nightly 19/01/2023),issue from one of [ these commits](https://ci.infra.status.im/job/status-mobile/job/nightly/2600/changes)

#### Expected behavior
can see the bottom sheet view, tap "View profile" and will be redirected to the user profile

#### Actual behavior
nothing happens on tapping "View profile"

https://user-images.githubusercontent.com/4557972/213772800-53b3b3d4-6fb1-4263-a868-2c53f6e7f161.mp4

### Reproduction
**Prerequisites:** user A and B are mutual contacts
1) Messages > Contacts
2) Tap on contact options
3) Tap on View Profile

### Additional Information

- Status version: nightly 20/01/2023
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by cammellos**
Adds support for compressed keys in:

1) User profile (old ui)
2) Contacts/other members profile (old ui)
3) Messages (new ui)

Fixes: #14843 
Fixes: #14812 

Removes pinned messages from profile as well, since it doesn't seem to be working and not needed from RC.

It's needed in combination of https://github.com/status-im/status-mobile/pull/14710 to have e2e tests running correctly , https://github.com/status-im/status-mobile/pull/14710#issuecomment-1398308912 for reference.

I don't think this handles compressed keys in universal links yet, that's probably something to do in a separate PR

### CODE REVIEW

File: src/status_im/multiaccounts/core.cljs
Comment (flexsurfer) [inline src/status_im/multiaccounts/core.cljs]: compressed-key should be shortened as well, in form like `1234...5678`, its compressed but not short i believe

File: src/status_im/profile/core.cljs
Comment (flexsurfer) [inline src/status_im/profile/core.cljs]: `:contacts/contact-built`, we're trying to have context in the event key

File: src/status_im/profile/core.cljs
Comment (flexsurfer) [inline src/status_im/profile/core.cljs]: extra cond?

File: src/status_im/ui/screens/profile/user/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/user/views.cljs]: same here for shorten

File: src/status_im/multiaccounts/core.cljs
Comment (cammellos) [reply src/status_im/multiaccounts/core.cljs]: It's ellipsed in the ui, I have checked (at least it looks like :) ), I think we use ellipse mode `middle`, we might want to have a custom method so the most relevant part is shown, but if that's ok, I'd do it in a separate PR

Comment (flexsurfer) [reply src/status_im/multiaccounts/core.cljs]: yeah but ellipsed didn't work well, and in some places we don't have it, so we made this util function, btw i believe it should just work , you don't need to make any changes , `(utils/get-shortened-address (or compressed-key public-key))`

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (ilmotta) [inline src/status_im/ui/screens/profile/contact/views.cljs:188]: Quick question: I see in the specs nodes SHOULD provide compression/decompression of public keys. Are we planning to eventually change the spec to force all nodes to have this capability?

**Review by ilmotta — APPROVED** [review]

**Review by ibrkhalil — APPROVED** [review]

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (cammellos) [reply src/status_im/ui/screens/profile/contact/views.cljs:188]: I don't think so, being a `MUST` is not necessary, since it's client dependent, for example if we started from compressed key from the get go, we would not need compression/decompression, and I can imagine soon enough we won't really need that, since the best way is to have status-go accept both format, and keep the conversion there, so it's transparent to clients and they can use them interchangeably.

Eventually, we could fade out the usage of the old key format and just index by the new key-id I suppose.

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (ilmotta) [reply src/status_im/ui/screens/profile/contact/views.cljs:188]: >keep the conversion there, so it's transparent to clients and they can use them interchangeably.

Understood, it does make sense :)

File: src/status_im/multiaccounts/core.cljs
Comment (erikseppanen) [inline src/status_im/multiaccounts/core.cljs:64]: I can see that this `contact-two-names` function was already accepting a `public-key?` parameter, so I wouldn't suggest refactoring this in this PR, but it does seem like `public-key?` is unnecessary since the fn could see if a public-key is passed in and decide accordingly what to return.

Nice use of FP in threading macro (not new, i know)

File: src/status_im/profile/core.cljs
Comment (erikseppanen) [inline src/status_im/profile/core.cljs:108]: If I were doing this, I'd use the name `show-profile-success` instead of `contact-built` because the caller might not be as interested in knowing the implementation details (that a contact was built), just that the call was successful.

**Review by erikseppanen — APPROVED** [review]

**Review by J-Son89 — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @cammellos Could you resolve conflicts, please?

**Comment by VolodLytvynenko** [comment]
Hi @cammellos Could you help me? I am confused a little bit in question about which chat key should be supported. In the current PR, only non-compressed chat keys are able to be used.  Did I understand correctly that both formats of the chat key should be able to be supported now?

**Comment by cammellos** [comment]
@VladimrLitvinenko This only shows compressed keys, it does not changing the way we handle them, so non-compressed keys still needs to be used, in conjunction with https://github.com/status-im/status-mobile/pull/14710 though, we should be able to use only compressed keys, hope that clear things up.

**Comment by VolodLytvynenko** [comment]
Hi @cammellos @pedro-et 
Should the current placeholder be changed to show that it expects to receive the format of a compressed chat key value? **(Example: z123..abc)** Currently, on design it looks like the non-compressed key is expected to be received
image

**Comment by cammellos** [comment]
> Hi @cammellos @pedro-et Should the current placeholder be changed to show that it expects to receive the format of a compressed chat key value? **(Example: z123..abc)** Currently, on design it looks like the non-compressed key is expected to be received image


Yes, probably in a separate PR though, as it's a small change and that part is till under development/review

**Comment by cammellos** [comment]
@VladimrLitvinenko any issue spotted that should be addressed in this PR?

**Comment by VolodLytvynenko** [comment]
> @VladimrLitvinenko any issue spotted that should be addressed in this PR?

Hi, @cammellos I did just a quick look at the build to prepare some test cases for myself. Did I understand correctly that this PR is finished and ready to be tested?

**Comment by cammellos** [comment]
@VladimrLitvinenko yes, this is ready to be tested, thank you

**Comment by VolodLytvynenko** [comment]
> @VladimrLitvinenko yes, this is ready to be tested, thank you

@cammellos Could you rebase it, please and resolve a conflicts?

**Comment by J-Son89** [comment]
@VladimrLitvinenko, according to the conversation in the issue https://github.com/status-im/status-mobile/issues/14886
(see the discussion at that link) this pr should fix https://github.com/status-im/status-mobile/issues/14843 and `14886`. 

Would it be possible to check that is the case and if so then we can close issues `14843` and `14886`

**Comment by VolodLytvynenko** [comment]
@J-Son89 Got it. Thank you!

**Comment by cammellos** [comment]
@VladimrLitvinenko rebased, thank you!

**Comment by VolodLytvynenko** [comment]
@cammellos Thank you for PR.  I didn't finish the tests yet. Just a quick update:

**ISSUE 1: The non-compressed chat key is shown instead of compressed one for profiles when the owner navigates to his profile**
**Steps to reproduce:**
1. Create new user
2. Open your profile page
3. Tap on the user '3 random name' to see the chat key

**Actual result:**
non-compressed chat key is shown
image

**Expected result:**
The compressed chat key is shown 

**Env:** Huawei p20 light, v9. iPhone 11 Pro Max v15.5

**Comment by cammellos** [comment]
@VladimrLitvinenko thanks for testing, apologies, I only tested after creation and login, missed that if you create an account/restore a key we still showed the old key.

It should be fixed now.

Thank you

**Comment by VolodLytvynenko** [comment]
> @VladimrLitvinenko thanks for testing, apologies, I only tested after creation and login, missed that if you create an account/restore a key we still showed the old key.
> 
> It should be fixed now.
> 
> Thank you

Hi. @cammellos  No problem. Let me know when it'll be ready for tests

**Comment by cammellos** [comment]
> > @VladimrLitvinenko thanks for testing, apologies, I only tested after creation and login, missed that if you create an account/restore a key we still showed the old key.
> > It should be fixed now.
> > Thank you
> 
> Hi. @cammellos No problem. Let me know when it'll be ready for tests

@VladimrLitvinenko it's ready for testing, thank you!

**Comment by VolodLytvynenko** [comment]
Hi @cammellos thank you for PR. It looks great. Only one low priority issue is found. it can be fixed as a separate follow up

**ISSUE 3: The non-compressed chat key is shown instead compressed within group details for owner of the group chat**
**Steps to reproduce:**
1. Create a group chat
2. Long tap on the created group -> tap group details

**Actual result:**
The owner of the group is shown with non-compressed chat key
image

**Expected result:**
The owner of the group is shown with compressed chat key

**Comment by VolodLytvynenko** [comment]
@cammellos Now we have current PR that supports the following:
- Compressed chat keys
-  Ability to open profile pages of contacts 

And this PR https://github.com/status-im/status-mobile/pull/14710 supports:
- New flow of searching contacts by their compressed chat key AND if the pasted compressed key is valid, THEN profile page of searched contact will be able to be opened (a feature that implemented in current https://github.com/status-im/status-mobile/pull/14845 PR)

Could you please cherry-pick this PR https://github.com/status-im/status-mobile/pull/14710 to the current? It will help to test their integrity. Then both can be merged if everything is ok

Thank you!

**Comment by cammellos** [comment]
> @cammellos Now we have current PR that supports the following:
> 
> * Compressed chat keys
> * Ability to open profile pages of contacts
> 
> And this PR #14710 supports:
> 
> * New flow of searching contacts by their compressed chat key AND if the pasted compressed key is valid, THEN profile page of searched contact will be able to be opened (a feature that implemented in current [Add support for compressed keys #14845](https://github.com/status-im/status-mobile/pull/14845) PR)
> 
> Could you please cherry-pick this PR #14710 to the current? It will help to test their integrity. Then both can be merged if everything is ok
> 
> Thank you!

Will do, I'll let you know when is ready, thanks

**Comment by cammellos** [comment]
> Hi @cammellos thank you for PR. It looks great. Only one low priority issue is found. it can be fixed as a separate follow up
> 
> **ISSUE 3: The non-compressed chat key is shown instead compressed within group details for owner of the group chat** **Steps to reproduce:**
> 
> 1. Create a group chat
> 2. Long tap on the created group -> tap group details
> 
> **Actual result:** The owner of the group is shown with non-compressed chat key image
> 
> **Expected result:** The owner of the group is shown with compressed chat key

@VladimrLitvinenko I have pushed and cherry picked the commit, the next build should include both changes.


Regarding Issue 3, if that's the only one and it tests successfully otherwise, I'd be inclined to addressed separately if that's ok with you, so we can get the bulk merged in.

Thanks

**Comment by VolodLytvynenko** [comment]
> @VladimrLitvinenko I have pushed and cherry picked the commit, the next build should include both changes.

@cammellos Thank you very much! Soon the feedback will be provided

> Regarding Issue 3, if that's the only one and it tests successfully otherwise, I'd be inclined to addressed separately if that's ok with you, so we can get the bulk merged in.
> 
> Thanks

Sure. I will create it as a separate follow up.

**Comment by VolodLytvynenko** [comment]
Hi @cammellos. Sorry for the delay in testing. I was waiting for the covering of this PR by automation script, but for today @churik is off. It will cause a lot of e2e failures if merge this PR now to the development branch without e2e coverage.
Please, don't merge it until Monday. I let you know.
Thanx!

**Comment by VolodLytvynenko** [comment]
Hi. @cammellos @erikseppanen After finishing the testing process, I noticed that some new commits were created in this PR. And current commits caused that compressed keys to be removed and again non-compressed are shown instead in the latest build.
Could you bring compressed keys back?

**Comment by cammellos** [comment]
> Hi. @cammellos @erikseppanen After finishing the testing process, I noticed that some new commits were created in this PR. And current commits caused that compressed keys to be removed and again non-compressed are shown instead in the latest build. Could you bring compressed keys back?

Apologies, I have wrongly rebased the build, fixed int the next build

**Comment by VolodLytvynenko** [comment]
> Apologies, I have wrongly rebased the build, fixed int the next build

@cammellos Got it. Retested it. Thank you for the hot fixes :) Please, Let the QA team know if some new commits will be created to retest it. Now, @churik is working on the creation of e2e scripts. Soon let you know when it can be merged

**Comment by churik** [comment]
fix for tests is here: https://github.com/status-im/status-mobile/pull/14930

**Comment by VolodLytvynenko** [comment]
hi, @cammellos thank you for your patience and for your work. E2E tests are ready and PR is able to be merged. Could you merge it, please?

**Comment by cammellos** [comment]
@VladimrLitvinenko on it, thank you for testing and your patience!

**Comment by VolodLytvynenko** [comment]
created a separate follow up https://github.com/status-im/status-mobile/issues/14934

**Pull request merged**
